packages/logger/src/Logger.ts

```diff
@@ -868,6 +876,22 @@ class Logger extends Utility implements LoggerInterface {
     extraInput: LogItemExtraInput
   ): void {
     if (logLevel >= this.logLevel) {
+      // Only flush buffer when log level is higher than the configured log level
+      if (logLevel > this.logLevel) {
+        const xRayTraceId = this.envVarsService.getXrayTraceId() as string;
```

Comment:
This can work in theory but in practice I don't see Powertools relying on this value to uniquely identify requests.

This value could be undefined in Lambda in rare occasions and would not exist at all outside of it, i.e. in testing environments or other compute platforms. It's true that we target primarily Lambda so this is not an immediate concern when it comes to other platforms, but introducing a requirement on this environment variable means now customers have to set it in their tests or CI, which is why for example the Tracer utility disables itself when running outside of Lambda.

Overall, for this type of thing I would instead relying on something that follows a similar principle but more robust, like [AsyncLocalStorage](https://nodejs.org/api/async_context.html#class-asynclocalstorage). With this being a language feature, we wouldn't have to worry about where Logger is running and context would be guaranteed at the process level.